operator/src/test/example-console.yaml

```diff
@@ -0,0 +1,12 @@
+apiVersion: console.streamshub.github.com/v1alpha1
+kind: Console
+metadata:
+  name: example
+  namespace: streams-console
+spec:
+  hostname: example-console.apps-crc.testing
+  kafkaClusters:
+    - kafkaUserName: console-kafka-user1
```

Comment:
does this refer to the name of a kafkauser resource?

Reply:
I see the answer to that is yes.  I don't think the field name conveys that well.

Reply:
I've been thinking about the name of this field a bit too and I'm not too happy with the name either. I haven't been able to come up with anything yet, do you have any ideas that make more sense?

Reply:
what about `kafkaUser_username`? It combines the kind and the field we need, something that can be easily retrieved running `kubectl get kafkauser -A -o jsonpath="{.items[*].status.username}"`

Reply:
This actually represents the `.metadata.name` of the `KafkaUser`. It's used to fetch the name of the generated secret from the status of the `KafkaUser`.

Reply:
`kafkaUser.metadata.name` then. Or `kafkaUser_metadata_name`. It might be verbose, but it's clear. And can be obtained easily `kubectl get kafkauser -A -o jsonpath="{.items[*].metadata.name}"  `

Reply:
If you're using Strimzi it's possible to have the KafkaUser CR in a different namespace to the Kafka CR, so what about:
```
kafkaUser:
   name: console-kafka-user1
   namespace: streams-console (optional)
```

In the Strimzi Kafka Access operator we actually use:
```
user:
    kind: KafkaUser
    apiGroup: kafka.strimzi.io
    name: my-user
    namespace: kafka
```
With the idea that in future we could support different types of user, we don't have a clear idea of what those would be, but just to make it easily extensible

Reply:
> If you're using Strimzi it's possible to have the KafkaUser CR in a different namespace to the Kafka CR, so what about:
> 
> ```
> kafkaUser:
>    name: console-kafka-user1
>    namespace: streams-console (optional)
> ```
> 
> In the Strimzi Kafka Access operator we actually use:
> 
> ```
> user:
>     kind: KafkaUser
>     apiGroup: kafka.strimzi.io
>     name: my-user
>     namespace: kafka
> ```
> 
> With the idea that in future we could support different types of user, we don't have a clear idea of what those would be, but just to make it easily extensible

I did not realize that. In that case, I liked what you've done in the access operator and we should probably do something like that here.

Reply:
(moved comment to thread)

> If you're using Strimzi it's possible to have the KafkaUser CR in a different namespace to the Kafka CR

@katheris can you confirm the above? I thought the `KafkaUser` must be in the same namespace as the `Kafka`, and are only operated on by the entity operator created for that `Kafka` CR.

Reply:
You specify an alternative namespace, see the docs https://strimzi.io/docs/operators/latest/deploying#deploying-the-user-operator-using-the-cluster-operator-str
```
You configure the entityOperator property of the Kafka resource to include the userOperator. By default, the User Operator watches for KafkaUser resources in the namespace of the Kafka cluster deployment. You can also specify a namespace using watchedNamespace in the User Operator spec. A single User Operator can watch a single namespace. One namespace should be watched by only one User Operator.
```

Reply:
Thanks, that helps. How about something like this where we can use a polymorphic `user` with a `type`. Initially we could support a `KafkaUser`, but it could be extended to other things like `Secret` later. wdyt?

```yaml
user:
  type: KafkaUser
  name: console-kafka-user1
  namespace: streams-console (optional)  
```

```yaml
user:
  type: Secret
  name: console-kafka-user1
  namespace: streams-console (optional)
  # both required
  usernameKey: <...>
  passwordKey: <...>
  # or alternatively use jaas config
  jaasConfigKey: <...>
```

Reply:
Yeah that looks reasonable to me

Reply:
It turns out that support for polymorphism in the tooling is not very good. I ended up going with this:

```yaml
credentials:
  # docs: Reference to a Strimzi KafkaUser resource
  user:
    name: console-kafka-user1
    namespace: streams-console (optional)
```

The expectation is that this could be expanded in the future to this, where exactly one of `user` or `secret` must be provided.

```yaml
credentials:
  # docs: Reference to a Strimzi KafkaUser resource
  user:
    name: console-kafka-user1
    namespace: streams-console (optional)
  secret:
    name: console-kafka-user1
    namespace: streams-console (optional)
    # both required
    usernameKey: <...>
    passwordKey: <...>
    # or alternatively use jaas config
    jaasConfigKey: <...>
```